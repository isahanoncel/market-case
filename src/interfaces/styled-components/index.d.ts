// #region Global Imports
import 'styled-components';
// #endregion Global Imports

export type Icons = 'Bag' | 'ArrowLeft';

export type Colors =
  | 'primary'
  | 'secondary'
  | 'white'
  | 'black100'
  | 'black600';

type Fonts = 'OpenSans';

declare module 'styled-components' {
  export interface DefaultTheme {
    colors: Record<Colors, string>;
    fonts: Record<Fonts, string>;
  }
}
