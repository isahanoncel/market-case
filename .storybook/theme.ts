import { create } from '@storybook/theming';

export default create({
  base: 'light',
  brandTitle: 'Getir',
  brandUrl: 'https://getir.com',
  brandTarget: '_self',
});
