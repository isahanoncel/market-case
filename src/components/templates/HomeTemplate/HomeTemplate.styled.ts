import styled from 'styled-components';
import { device } from '../../../constants/styles/GlobalStyle';

export const HomeTemplateWrapper = styled.div`
  width: 100%;
  height: 100%;
  min-height: 100vh;
  background-color: ${({ theme }) => theme.colors.background};

  ${device.mobile} {
    margin-top: 80px;
  }
`;
export const HomeTemplateContentWrapper = styled.div`
  display: flex;
  flex-wrap: wrap;
  gap: 16px;
  margin-top: 40px;

  ${device.mobile} {
    display: block;
    padding: 10px 16px;
  }
`;
export const HomeTemplateContentBox = styled.div`
  flex: 1;
`;

export const HeaderLink = styled.a`
  margin: auto;
`;
