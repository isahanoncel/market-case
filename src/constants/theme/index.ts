import { DefaultTheme } from 'styled-components';

export const COLORS = {
  white: '#fff',
  primary: '#1EA4CE',
  whiteBlue: '#F2F0FD',
  secondary: '#147594',
  black100: '#E0E0E0',
  black500: '#6F6F6F',
  black600: '#525252',
  gray: '#697488',
  background: '#E5E5E5',
  transparent: 'transparent',
  innerBorder: '#DFDEE2',
  thinGray: '#A8A8A8',
};

export const FONTS = {
  OpenSans: "'Open Sans', sans-serif",
};

const heroTheme: DefaultTheme = {
  colors: COLORS,
  fonts: FONTS,
};

export default heroTheme;
