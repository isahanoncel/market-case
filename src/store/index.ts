import { configureStore } from '@reduxjs/toolkit';
import { useDispatch, useSelector, TypedUseSelectorHook } from 'react-redux';
import products from './slices/products';
import basket from './slices/basket';
import filter from './slices/filter';

const store = configureStore({
  reducer: {
    products,
    basket,
    filter,
  },
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;

export const useAppDispatch = () => useDispatch<AppDispatch>();
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;

export default store;
