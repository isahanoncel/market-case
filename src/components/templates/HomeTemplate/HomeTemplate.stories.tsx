import React from 'react';
import { ComponentMeta, ComponentStory } from '@storybook/react';
import { Provider } from 'react-redux';
import HomeTemplate from '.';
import store from '../../../store';

export default {
  title: 'Templates/HomeTemplate',
  component: HomeTemplate,
} as ComponentMeta<typeof HomeTemplate>;

export const Default: ComponentStory<typeof HomeTemplate> = () => (
  <Provider store={store}>
    <HomeTemplate />
  </Provider>
);
