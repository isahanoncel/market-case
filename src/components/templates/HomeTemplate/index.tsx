import React, { useEffect, useLayoutEffect, useState } from 'react';
import { ICompany } from '../../../interfaces/company';
import CompanyService from '../../../service/company';
import ProductService from '../../../service/product';
import { useAppDispatch } from '../../../store';
import { setProducts } from '../../../store/slices/products';
import Container from '../../atoms/Container';
import BasketList from '../../molecules/BasketList';
import Footer from '../../molecules/Footer';
import Filters from '../../organisms/Filters';
import Header from '../../organisms/Header';
import ProductListing from '../../organisms/ProductListing';
import * as S from './HomeTemplate.styled';

const HomeTemplate = () => {
  const [companies, setCompanies] = useState<ICompany[]>([]);
  const [tags, setTags] = useState<string[]>([]);
  const [showBasketList, setShowBasketList] = useState<boolean>(false);
  const [productRequestLoading, setProductRequestLoading] = useState(true);
  const dispatch = useAppDispatch();

  useLayoutEffect(() => {
    setShowBasketList(window.innerWidth > 765);
  }, []);

  useEffect(() => {
    CompanyService.getCompanies().then((data) => {
      setCompanies(data);
    });

    ProductService.getTags().then((data) => {
      setTags(data);
    });

    ProductService.getProducts()
      .then((data) => {
        dispatch(setProducts(data));
      })
      .finally(() => {
        setProductRequestLoading(false);
      });
  }, []);

  return (
    <S.HomeTemplateWrapper>
      <Header setShowBasketList={setShowBasketList} />
      <Container>
        <S.HomeTemplateContentWrapper>
          <S.HomeTemplateContentBox>
            <Filters tags={tags} companies={companies} />
          </S.HomeTemplateContentBox>
          <S.HomeTemplateContentBox>
            <ProductListing isLoading={productRequestLoading} />
          </S.HomeTemplateContentBox>
          <S.HomeTemplateContentBox>
            {showBasketList && <BasketList />}
          </S.HomeTemplateContentBox>
        </S.HomeTemplateContentWrapper>
        <Footer />
      </Container>
    </S.HomeTemplateWrapper>
  );
};

export default HomeTemplate;
