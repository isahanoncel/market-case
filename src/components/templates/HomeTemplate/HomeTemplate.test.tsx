import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import HomeTemplate from '.';
import theme from '../../../constants/theme';
import store from '../../../store';

test('HomeTemplate snapshot testing', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <ThemeProvider theme={theme}>
          <HomeTemplate />
        </ThemeProvider>
      </Provider>,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
