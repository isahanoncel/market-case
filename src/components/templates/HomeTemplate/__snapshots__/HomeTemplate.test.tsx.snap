// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeTemplate snapshot testing 1`] = `
.c2 {
  max-width: 1232px;
  margin: 0 auto;
  height: 100%;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: 24px 0 0 0;
}

.c17 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c6 {
  font-size: 14px;
  line-height: 14px;
  color: #fff;
  font-weight: 600;
  margin: 0;
}

.c10 {
  font-size: 13px;
  line-height: 14px;
  color: #697488;
  font-weight: 600;
  margin: 0;
}

.c14 {
  font-size: 14px;
  line-height: 18px;
  color: #525252;
  font-weight: 400;
  margin: 0;
}

.c11 {
  background-color: #fff;
  box-shadow: 0px 6px 24px rgba(93,62,188,0.04);
  border-radius: 2px;
  margin-top: 12px;
  padding: 24px;
  max-height: 184px;
  overflow-y: scroll;
}

.c12 {
  cursor: pointer;
  margin-bottom: 18px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c12:last-of-type {
  margin-bottom: 0;
}

.c13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 22px;
  height: 22px;
  border: 2px solid #DFDEE2;
  border-radius: 22px;
  margin: 0 12px 0 0;
}

.c16 {
  margin-bottom: 17px;
}

.c18 {
  font-size: 14px;
  outline: none;
  padding: 12px 16px;
  color: #525252;
  border: 2px solid #E0E0E0;
  border-radius: 2px;
}

.c18::-webkit-input-placeholder {
  color: #E0E0E0;
}

.c18::-moz-placeholder {
  color: #E0E0E0;
}

.c18:-ms-input-placeholder {
  color: #E0E0E0;
}

.c18::placeholder {
  color: #E0E0E0;
}

.c15 {
  background-color: #fff;
  box-shadow: 0px 6px 24px rgba(93,62,188,0.04);
  border-radius: 2px;
  margin-top: 12px;
  padding: 24px;
  max-height: 244px;
  overflow: hidden;
}

.c19 {
  max-height: 131px;
  overflow-y: scroll;
}

.c5 {
  background-color: #147594;
  padding: 30px 24px;
  text-align: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c5 svg {
  margin-right: 13.41px;
}

.c1 {
  width: 100%;
  background-color: #1EA4CE;
}

.c4 {
  margin: auto;
}

.c0 {
  width: 100%;
  height: 100%;
  min-height: 100vh;
  background-color: #E5E5E5;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  gap: 16px;
  margin-top: 40px;
}

.c8 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

@media (min-width:0px) and (max-width:768px) {
  .c3 {
    height: auto;
  }
}

@media (min-width:0px) and (max-width:768px) {
  .c9 {
    height: auto;
  }
}

@media (min-width:0px) and (max-width:768px) {
  .c17 {
    height: auto;
  }
}

@media (min-width:0px) and (max-width:768px) {
  .c1 {
    position: fixed;
    top: 0;
  }
}

@media (min-width:0px) and (max-width:768px) {
  .c0 {
    margin-top: 80px;
  }
}

@media (min-width:0px) and (max-width:768px) {
  .c7 {
    display: block;
    padding: 10px 16px;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <div
        className="c3"
      >
        <a
          className="c4"
          href="/"
        >
          <img
            alt="Getir"
            height={40}
            src="/assets/logo.svg"
            width={141}
          />
        </a>
        <div
          className="c5"
        >
          <svg
            fill="#1EA4CE"
            height={16}
            viewBox="0 0 16 16"
            width={16}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0.411743 5.46655H13.5884V17.0081H0.411743V5.46655Z"
              fill="white"
            />
            <path
              d="M4.67188 0.657471H9.34121L10.2765 1.62625V5.4666L9.26455 5.46388V1.6242H4.75454V5.46388L3.72388 5.4666V1.6242L4.67188 0.657471Z"
              fill="white"
            />
          </svg>
          <span
            className="c6"
            color="white"
            size={14}
          >
            â‚º 
            0
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="c2"
  >
    <div
      className="c7"
    >
      <div
        className="c8"
      >
        <div
          className="c9"
        >
          <span
            className="c10"
            color="gray"
            size={13}
          >
            Sorting
          </span>
          <div
            className="c11"
          >
            <div
              className="c12"
              onClick={[Function]}
            >
              <div
                checked={false}
                className="c13"
              />
              <span
                className="c14"
                color="black600"
                size={14}
              >
                Price low to high
              </span>
            </div>
            <div
              className="c12"
              onClick={[Function]}
            >
              <div
                checked={false}
                className="c13"
              />
              <span
                className="c14"
                color="black600"
                size={14}
              >
                Price high to low
              </span>
            </div>
            <div
              className="c12"
              onClick={[Function]}
            >
              <div
                checked={false}
                className="c13"
              />
              <span
                className="c14"
                color="black600"
                size={14}
              >
                New to old
              </span>
            </div>
            <div
              className="c12"
              onClick={[Function]}
            >
              <div
                checked={false}
                className="c13"
              />
              <span
                className="c14"
                color="black600"
                size={14}
              >
                Old to new
              </span>
            </div>
          </div>
        </div>
        <div
          className="c9"
        >
          <span
            className="c10"
            color="gray"
            size={13}
          >
            Brands
          </span>
          <div
            className="c15"
          >
            <div
              className="c16"
            >
              <div
                className="c17"
              >
                <input
                  className="c18"
                  onChange={[Function]}
                  placeholder="Search Brands"
                  type="text"
                />
              </div>
            </div>
            <div
              className="c19"
            />
          </div>
        </div>
        <div
          className="c9"
        >
          <span
            className="c10"
            color="gray"
            size={13}
          >
            Tags
          </span>
          <div
            className="c15"
          >
            <div
              className="c16"
            >
              <div
                className="c17"
              >
                <input
                  className="c18"
                  onChange={[Function]}
                  placeholder="Search Tags"
                  type="text"
                />
              </div>
            </div>
            <div
              className="c19"
            />
          </div>
        </div>
      </div>
      <div
        className="c8"
      >
        Products
      </div>
      <div
        className="c8"
      >
        Basket
      </div>
    </div>
  </div>
</div>
`;
